### PR TITLE
Create topics-collection landing page

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Created new template for `topics-collections` that uses `hugo` frontmatter to set custom content to display on a new `topics-collection` landing page.

### Preview

- [New Topic Collection](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-topics-collection-template/topics/design)
- [Current Topic Standard](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-topics-collection-template/topics/agile)

### Solution

Instead of using hugo page obects, front matter from the topic markdown page is used to set the curated content on the page.

1. Created `topics-collection` landing page — adb4491e102c7e1a17e9149c7cef95a92a15b062
2. Front matter data to display curated guides, communities, and resources — 009ae16e55589106d671774ce076320876232a9b
3. Several updates to the `collection` component to handle the custom front matter for displaying lists — a38be4af5cd0837312c607afdbf955fb9de4d850, 3b2373c5b2ab1f6a52f47d4747b7a0a9b750ea1d 

**Single Resource & Community**
Screen Shot 2023-09-20 at 11 34 36 AM

**Multiple Resource & Community**
Screen Shot 2023-09-20 at 10 12 28 AM

### How To Test

1. First Step
2. Second Step
3. Third Step

---

### Dev Checklist

- [ ] PR has correct labels
- [ ] A11y testing (voice over testing, meets WCAG, run axe tools)
- [ ] Code is formatted properly

### CODE REVIEW

File: content/topics/design/_index.md [hunk 1]
```diff
@@ -7,13 +7,51 @@ slug: "design"
...
+
+# populate the topic-collection template with featured links
+data:
+  header:
+    title: "Design"
+    deck: "Understand how and why design impacts user experience"
+  contents:
+    summary: "Guidance, resources, and community to help you use design and user experience to create government websites that meet customer needs, work well on any device, and follow federal web requirements."
+    legislation:
+      name: "21 Century IDEA"
```
Comment (bonnieAcameron) [inline content/topics/design/_index.md:24]:
```suggestion
      name: "21st Century Integrated Digital Experience Act (21st Century IDEA)"
```

File: content/topics/design/_index.md [hunk 2]
```diff
@@ -7,13 +7,51 @@ slug: "design"
...
+      description: ""
+      link_text: "View Policy"
+      image: "hcd-discovery-operations.png"
+      link_url: "https://www.govinfo.gov/app/details/DCPD-202101050"
+  featured_resources:
+      multiple: false
+      resources:
+      - link: "/guides/hcd/discovery-concepts"
+        kicker: "HCD Guides"
```
Comment (bonnieAcameron) [inline content/topics/design/_index.md:33]:
```suggestion
        kicker: "Human-Centered Design Guide Series"
```

File: content/topics/design/_index.md [hunk 3]
```diff
@@ -7,13 +7,51 @@ slug: "design"
...
+  communities:
+      multiple: false
+      community:
+        - link: "/communities/user-experience"
+  top_resources_header: "Essential knowledge for design"
```
Comment (bonnieAcameron) [inline content/topics/design/_index.md:38]:
```suggestion
  top_resources_header: "Design: essential knowledge"
```

File: themes/digital.gov/layouts/partials/core/topics/topics-collection.html [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+      {{/* Merges the future and past events into one set */}}
+      {{- $events_stream := union $future_events $past_events -}}
+
+      {{/* Sorting all the items by date and reverse chron */}}
+      {{- $events_stream := $events_stream.ByDate.Reverse -}}
+
+      {{/* If there are any events */}}
+      {{- if $events_stream -}}
+        <h2>Events for {{ .Title }}</h2>
```
Comment (bonnieAcameron) [inline themes/digital.gov/layouts/partials/core/topics/topics-collection.html:135]:
```suggestion
        <h2>{{ .Title }} events</h2>
```

File: themes/digital.gov/layouts/partials/core/topics/topics-collection.html [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    <div class="grid-row grid-container grid-container-desktop">
+      {{/* Gets the recent blog posts */}}
+      {{- $posts := where .Paginator.Pages "Section" "news" -}}
+
+      {{/* Sorting all the items by date and reverse chron */}}
+      {{- $posts_stream := $posts.ByDate.Reverse -}}
+
+      {{/* If there are any events */}}
+      {{- if $posts_stream -}}
+        <h2>News on {{ .Title -}}</h2>
```
Comment (bonnieAcameron) [inline themes/digital.gov/layouts/partials/core/topics/topics-collection.html:161]:
```suggestion
        <h2>{{ .Title -}} news</h2>
```

**Review by bonnieAcameron — CHANGES_REQUESTED** [review]
Looks good, Nick! I added my comments. I have a few design ideas that are small. I will capture them in another comment.

**Comment by bonnieAcameron** [comment]
Design comments:
Overall lovely!

We may want to address: 

- a different style applied to "join the community" than to a "guide". Maybe the dg-ring for "join the community"?
- I think I want the image on the policy card to look a little more "official." example incoming:

**Comment by bonnieAcameron** [comment]
@nick-mon1 here is the image we chose: 
legal-judicial-bro-vector-istock-getty-images-1388107868

**Comment by nick-mon1** [comment]
@bonnieAcameron I like the idea of using something different for the 'communities' block, the `ring` shortcode doesn't take in links. It's more of a glorified text box. We can discuss better ideas for this.

**Comment by ToniBonittoGSA** [comment]
@nick-mon1 
In the [2nd preview link](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-topics-collection-template/topics/agile/), `News and Events on Agile` should be styled the same as `Resources on Agile` and `Join a Community of Practice`. 

Last copy in the Wayback Machine, June 2023: https://web.archive.org/web/20230601145635/https://digital.gov/topics/agile/ 

Thx!

**Review by bonnieAcameron — DISMISSED** [review]
Looks great!

**Review by mejiaj — DISMISSED** [review]

**Review by ToniBonittoGSA — DISMISSED** [review]

**Pull request merged**